tests/levenberg_marquardt_test.py

```diff
@@ -185,7 +184,7 @@ def test_double_exponential_fit_x32(self, materialize_jac, geodesic):
   # Optimization and Data Fitting" Example 6.10-. page 124. Data taken from
   # immoptibox "A MATLAB TOOLBOX FOR OPTIMIZATION AND DATA FITTING" H.B. Nielsen
   @unittest.skipIf(not jax.config.jax_enable_x64, 'test requires X64')
```

Comment:
Add jax.config.update("jax_enable_x64", True) at the start and jax.config.update("jax_enable_x64", False) at the end to be sure that the test is done.

Reply:
I also thought the same, and that does work for the test internal code, however, apparently, `@unittest.skipIf(...)` uses the module level environmental variables, which means that changing the setting as part of the test will not take into effect, therefore one would need to do something like the following to enforce double precision FP:

```
JAX_ENABLE_X64=True python levenberg_marquardt_test.py
```

Reply:
I was thinking of simply removing @unittest.skipIf(not jax.config.jax_enable_x64, 'test requires X64') in favor of adding jax.config.update("jax_enable_x64", True) at the start and jax.config.update("jax_enable_x64", False) at the end. I am not sure that the github tests for example are configured to run both in x64 and x32.

Reply:
I'm also not sure how is best to do this. If we go with what you mentioned, we are basically ignoring all x32 tests (we might as well just remove them in that situation). That was actually my early on setup that I remember @mblondel suggested that we need to add x32 as it is the default usage of jaxopt. In the current setup, we are also ignoring x64 for most of the automatic integration tests, however, running x64 only needs a user specification side rather than changing the code.

Reply:
Ok, my bad, I didn't realize the presence of the x32 test. Could you then simply make sure that `test_double_exponential_fit_x32` is also testing for different solvers?

Reply:
Interestingly, if we use QR solver, the algorithm doesn't converge for the x32 test. I added all the other solvers and they are fine, also fine for x64.

Reply:
Also, I updated the code in the materialize Jacobian to use the expected solvers and therefore removed the previous TODO and only skipping the combination that fails.

Reply:
Ok, we may want to end this PR, so I am ok to add a todo and a warning and add an enhancement issue to review this later. We should also start having benchmarks.